nbgrader/nbextensions/assignment_list/assignment_list.js

```diff
@@ -413,7 +432,15 @@ define([
                     },
                     type : "POST",
                     dataType : "json",
-                    success : $.proxy(that.on_refresh, that),
+                    success : function (data, status, xhr) {
+                        if (!data.success) {
+                            that.submit_error(data);
+                            button.text('Sumbit');
+                            button.removeAttr('disabled');
+                        }
+                        //$.proxy(that.on_refresh, that);
+                        //GRRRR
```

Comment:
I *think* you should be able to do `that.on_refresh()`, rather than using the `$.proxy`. The way you're doing it here you're creating a closure with this function you've defined -- which is sort of what `$.proxy` is doing in the first place. JavaScript is confusing and I don't entirely understand why that's necessary, but if you don't create the closure (e.g. `success : that.on_refresh`), then when the function is called it won't be associated with the relevant instance of the class. So that's why the `$.proxy` is there to begin with, but I think if you remove it here then it shouldn't be a problem.